system_baseline/views/v1.py

```diff
@@ -54,10 +51,14 @@ def get_baselines_by_ids(baseline_ids, limit, offset, order_by, order_how):
     query = SystemBaseline.query.filter(
         SystemBaseline.account == account_number, SystemBaseline.id.in_(baseline_ids)
     )
-    try:
-        ensure_correct_system_count(baseline_ids, query.all())
-    except ItemNotReturned as e:
-        raise HTTPError(HTTPStatus.NOT_FOUND, message=e.message)
+    full_results = query.all()
+    if len(full_results) < len(baseline_ids):
+        fetched_ids = {result.id for result in full_results}
+        missing_ids = set(baseline_ids) - fetched_ids
```

Comment:
this is comparing UUID objects coming back from the baseline object to strings in the URL param and failing. You probably want this:

```python
        fetched_ids = {str(result.id) for result in full_results}
```

otherwise it will compare UUID('whatever') to 'whatever', and even if 'whatever' exists, it will be in the error message.